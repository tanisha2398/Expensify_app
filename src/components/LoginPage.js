import React from "react";
import { connect } from "react-redux";
import { startLogin, startFacebookLogin } from "../actions/auth";
export const LoginPage = ({ startLogin, startFacebookLogin }) => {
  return (
    <div className="box-layout">
      <div className="box-layout__box">
        <h1 className="box-layout__title">Expensify</h1>
        <p>Its time to get your expenses under control</p>
        <button className="button" onClick={startLogin}>
          Login with Google
        </button>
        <button
          className="button button--facebook"
          onClick={startFacebookLogin}
        >
          Login with Facebook
        </button>
      </div>
    </div>
  );
};

const mapDispatchToProps = dispatch => ({
  startLogin: () => dispatch(startLogin()),
  startFacebookLogin: () => dispatch(startFacebookLogin())
});

export default connect(
  undefined,
  mapDispatchToProps
)(LoginPage);
