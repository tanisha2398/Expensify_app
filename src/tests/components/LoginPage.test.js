import React from "react";
import { shallow } from "enzyme";
import LoginPage from "../../components/LoginPage";

test("should render login correctly", () => {
  const wrapper = shallow(<LoginPage />);
  expect(wrapper).toMatchSnapshot();
});
