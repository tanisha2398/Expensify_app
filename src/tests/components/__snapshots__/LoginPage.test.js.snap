// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render login correctly 1`] = `
<div>
  <button>
    Login
  </button>
</div>
`;
